src/basics.ml

```diff
@@ -240,6 +240,16 @@ let distinct_vars : term array -> bool =
     | _ -> raise Not_unique_var
   in try Array.iter check ts; true with Not_unique_var -> false
 
+(** [is_pattern t] tells whether term [t] is a valid pattern. *)
+let rec is_pattern : term -> bool = fun t ->
+  match get_args (unfold t) with
+  | (Patt(_), []  ) -> true (* Pattern variables are not applied *)
+  | (Patt(_), args) -> distinct_vars (Array.of_list args)
```

Comment:
should be `| (Patt(_,_,ts), []) -> distinct_vars ts`

Reply:
This covers the case `&F[x, y]`, but not the case `&F x y` which apparently can happen.